### PR TITLE
Use results summary az dev api

### PR DESCRIPTION
**Pull request by alexperovich**

### CODE REVIEW

**Comment by alexperovich** [comment]
@JunTaoLuo this change will fix the flaky test issue

File: src/Microsoft.DotNet.Helix/Sdk/CheckAzurePipelinesTestResults.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                return;
+            }
+            var data = await RetryAsync(
+                async () =>
+                {
+                    using (var req = new HttpRequestMessage(
+                        HttpMethod.Get,
+                        $"{CollectionUri}{TeamProject}/_apis/test/resultsummarybybuild?buildId={BuildId}&api-version=5.1-preview.2"))
+                    {
+                        using (var res = await client.SendAsync(req))
```
Comment (ChadNedzlek) [inline src/Microsoft.DotNet.Helix/Sdk/CheckAzurePipelinesTestResults.cs]: Lots of vars. :-)

**Comment by JunTaoLuo** [comment]
Any updates on this PR? It looks approved and all tests are passing.

**Comment by alexperovich** [comment]
I was waiting for a good build. I just am not looking at this PR closely. I'll merge it now.

**Pull request merged**